### PR TITLE
database/raft: enforce allowed members list

### PR DESCRIPTION
**Pull request by tessr**
Enforces the allowed members list and adds an endpoint to actually add to this list. 

Next: Connecting corectl to that endpoint.

### CODE REVIEW

**Comment by tessr** [comment]
PTAL. Could use some feedback on the http request/response bodies especially.

File: database/raft/raft.go [hunk 1]
```diff
@@ -175,6 +175,7 @@ func Start(laddr, dir, bootURL string, httpClient *http.Client, useTLS bool) (*S
 
 	// TODO(kr): grpc
 	sv.mux.HandleFunc("/raft/join", sv.serveJoin)
+	sv.mux.HandleFunc("/raft/add-allowed-member", sv.serveAddAllowedMember)
```
Comment (kr) [inline database/raft/raft.go]: Is this gonna need to move out of this package so it can also edit the grants list?

File: database/raft/raft.go [hunk 2]
```diff
@@ -627,6 +654,15 @@ func (sv *Service) join(addr, baseURL string) error {
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		errmsg, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return errors.Wrap(err, "could not parse response from boot server")
+		}
+		defer resp.Body.Close()
+		return errors.New(fmt.Sprintf("boot server responded with status %d: %s", resp.StatusCode, errmsg))
```
Comment (kr) [inline database/raft/raft.go]: This can use fmt.Errorf.

**Comment by kr** [comment]
> Could use some feedback on the http request/response bodies especially.

The look fine to me for 1.2.

File: database/raft/raft.go (hunk 2 above)
Comment (tessr) [inline database/raft/raft.go]: _omg_ I couldn't remember what that function was, I spent about a minute staring at my screen trying to remember. 😳

**Comment by tessr** [comment]
Ready for another pass, PTAL!

File: core/membership.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package core
+
+import (
+	"context"
+
+	"chain/errors"
+)
+
+var errMissingAddr = errors.New("missing address")
```
Comment (kr) [inline core/membership.go]: Did you mean to put this in errors.go as a 400?

**Comment by kr** [comment]
LGTM (with one minor question)

**Pull request merged**